### PR TITLE
feat: add information page (distribute assessment - page 3)

### PR DESCRIPTION
**Pull request by joyce-shi**
## Notion ticket link

[Add Information page](https://www.notion.so/uwblueprintexecs/Add-Information-page-Distribute-Assessment-Page-3-d1d072bb0b564789876a22479ecd9672?pvs=4)

## Implementation description
* Implemented 3rd flow in the Distribute Assessment flow

## Steps to test
1. UI renders as expected
2. Performs the appropriate form validation

### CODE REVIEW

File: frontend/src/components/teacher/session-creation/steps/AddInformation.tsx [hunk 1]
```diff
@@ -1,22 +1,99 @@
...
   setEndDate,
   setNotes,
+  setValidDates,
 }: AddInformationProps): React.ReactElement => {
+  const [invalidStartDate, setInvalidStartDate] = useState(false);
+  const [invalidEndDate, setInvalidEndDate] = useState(false);
+  const hasPast = (input: Date) => {
+    const now = new Date();
+    return input < now && !isSameDay(input, now);
+  };
```
Comment (carissa-tang) [inline frontend/src/components/teacher/session-creation/steps/AddInformation.tsx]: move this to utils?

Comment (joyce-shi) [reply frontend/src/components/teacher/session-creation/steps/AddInformation.tsx]: good idea! done :D

**Review by carissa-tang — COMMENTED** [review]
can you double check this case to remove the date error

1. choose start > end > current
2. choose start s.t. end > start > current

https://github.com/uwblueprint/jump-math/assets/69697180/eb535fd7-ebe0-41cd-8b2e-32598b18b9fe

**Comment by joyce-shi** [comment]
> can you double check this case to remove the date error
> 
> 1. choose start > end > current
> 2. choose start s.t. end > start > current
> 
>  Screen.Recording.2023-07-21.at.9.54.03.PM.mov

good catch! can you test this feature again? 🙏

File: frontend/src/components/pages/teacher/DisplayAssessmentsPage.tsx [hunk 2]
```diff
@@ -98,7 +98,7 @@ const DisplayAssessmentsPage = (): React.ReactElement => {
           <ErrorState />
         </Box>
       )}
-      {!!formattedData?.length && !loading && !error && (
+      {!!formattedData?.length && !loading && error && (
```
Comment (carissa-tang) [inline frontend/src/components/pages/teacher/DisplayAssessmentsPage.tsx]: this appeared again

File: frontend/src/components/teacher/session-creation/steps/AddInformation.tsx [hunk 3]
```diff
@@ -1,22 +1,90 @@
...
+    if (startDate) {
+      setInvalidStartDate(isPastDate(startDate));
+    }
+    if (endDate) {
+      setInvalidEndDate(isPastDate(endDate));
+    }
+    if (startDate && endDate) {
+      setInvalidStartDate(isPastDate(startDate) || endDate <= startDate);
+      setInvalidEndDate(isPastDate(endDate) || endDate <= startDate);
+    }
```
Comment (carissa-tang) [inline frontend/src/components/teacher/session-creation/steps/AddInformation.tsx]:
```suggestion
    if (startDate && endDate) {
      setInvalidStartDate(isPastDate(startDate) || endDate <= startDate);
      setInvalidEndDate(isPastDate(endDate) || endDate <= startDate);
      return;
    } 

    if (startDate) {
      setInvalidStartDate(isPastDate(startDate));
    } 
    if (endDate) {
      setInvalidEndDate(isPastDate(endDate));
    }
```
reorder so we don't have to recompute?

File: frontend/src/components/pages/teacher/DistributeAssessmentPage.tsx [hunk 4]
```diff
@@ -34,6 +34,8 @@ const DistributeAssessmentPage = (): React.ReactElement => {
   const [endDate, setEndDate] = useState<Date | null>(null);
   const [notes, setNotes] = useState("");
 
+  const [validDates, setValidDates] = useState(true);
```
Comment (carissa-tang) [inline frontend/src/components/pages/teacher/DistributeAssessmentPage.tsx]: it looks like we're not setting this value. should we default false and then update value in the useEffect? or we can use `invalidStartDate` and `invalidEndDate` directly

File: frontend/src/components/teacher/session-creation/steps/AddInformation.tsx (hunk 3 above)
Comment (joyce-shi) [reply frontend/src/components/teacher/session-creation/steps/AddInformation.tsx:47]: good catch

File: frontend/src/components/pages/teacher/DistributeAssessmentPage.tsx (hunk 4 above)
Comment (joyce-shi) [reply frontend/src/components/pages/teacher/DistributeAssessmentPage.tsx]: updated to set the value in the hook

**Review by carissa-tang — APPROVED** [review]

**Pull request merged**